File: zig/src/types/StatusInfo.zig
const std = @import("std");

pub const StatusInfo = struct {
    branch: []const u8,
    up_to_date: []const u8,
    staged: []const []const u8,
    modified: []const []const u8,
    untracked: []const []const u8,
    deleted: []const []const u8,

    pub fn deinit(self: StatusInfo, allocator: std.mem.Allocator) void {
        allocator.free(self.branch);
        allocator.free(self.up_to_date);

        for (self.staged) |item| allocator.free(item);
        allocator.free(self.staged);

        for (self.modified) |item| allocator.free(item);
        allocator.free(self.modified);

        for (self.untracked) |item| allocator.free(item);
        allocator.free(self.untracked);

        for (self.deleted) |item| allocator.free(item);
        allocator.free(self.deleted);
    }
};

File: dotnet/Gitologist/Types/StatusInfo.cs
namespace Gitologist.Types;

public record class StatusInfo
{
    public required string Branch { get; init; }
    public required string UpToDate { get; init; }
    public required string[] Staged { get; init; }
    public required string[] Modified { get; init; }
    public required string[] Untracked { get; init; }
    public required string[] Deleted { get; init; }
}
